pkg/registry/core/service/ipallocator/controller/repair.go

```diff
@@ -80,17 +75,39 @@ func NewRepair(interval time.Duration, serviceClient corev1client.ServicesGetter
 	eventBroadcaster.StartRecordingToSink(&corev1client.EventSinkImpl{Interface: eventClient.Events("")})
 	recorder := eventBroadcaster.NewRecorder(legacyscheme.Scheme, v1.EventSource{Component: "ipallocator-repair-controller"})
 
+	// build *ByFamily struct members
```

Comment:
What about feature gate?  Shouldn't this all be neutralized if the gate is closed?  It may be easier to fork the original functions and switch at the top for gate-enabled vs gate-disabled.  Harder to review, but easier to be confident in.

Reply:
as agreed we will gate the allocator but not the repair logic.